src/System.Diagnostics.StackTrace/src/System/Diagnostics/StackTrace.netcore50aot.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using Internal.Diagnostics;
+
+namespace System.Diagnostics
+{
+    public sealed class StackTrace
+    {
+        private readonly IntPtr[] _ips;
+        private readonly bool _fNeedFileInfo;
+
+        public StackTrace(Exception exception, bool needFileInfo)
+        {
+            if (exception == null)
+            {
+                throw new ArgumentNullException("exception");
+            }
+            _ips = ExceptionExtensions.GetStackIPs(exception);
+            _fNeedFileInfo = needFileInfo;
+        }
+
+        public StackFrame[] GetFrames()
+        {
+            // There are special IPs mixed in to the list that aren't intended to be exposed as StackFrames
+            // Deliberately doing this without Linq to keep the dependency set down.
+            StackFrame[] tempFrames = new StackFrame[_ips.Length];
+            int countNonSpecialIPs = 0;
+            for (int i = 0; i < _ips.Length; i++)
```

Comment:
I'm planning to leave this as-is. Even if we assumed the proposed change saved 10k cpu cycles (which I suspect is extremely generous) it still is unlikely to move the needle on the overall performance for scenarios hitting this code path. We can easily revisit if anyone produces perf data showing this is an issue.